src/main/java/org/reaktivity/nukleus/maven/plugin/internal/ast/parse/AstParser.java

```diff
@@ -863,6 +867,35 @@ public Builder visitList_member(
             return super.visitNon_primitive_member_with_default(ctx);
         }
 
+        @Override
+        public Builder visitVariant_array(
+            Variant_arrayContext ctx)
+        {
+            listMemberBuilder.name(ctx.declarator(2).getText());
+            String variantArrayTypeName = ctx.declarator(0).getText();
+            AstType astVariantArrayTypeName = Objects.requireNonNullElse(astTypesByQualifiedName.get(variantArrayTypeName),
+                lookUpAstType(variantArrayTypeName));
+            listMemberBuilder.type(astVariantArrayTypeName);
+
+            String arrayItemTypeName = ctx.declarator(1).getText();
+            AstType astArrayItemTypeName = Objects.requireNonNullElse(astTypesByQualifiedName.get(arrayItemTypeName),
+                lookUpAstType(arrayItemTypeName));
+            listMemberBuilder.arrayTypeName(astArrayItemTypeName);
+            listMemberBuilder.arrayType(AstType.VARIANT);
+            return super.visitVariant_array(ctx);
```

Comment:
IIRC, you can name the declarators in the grammar to avoid extracting them by index.
Then if the grammar changes later, it will be more resilient in cases where the declarator index is inadvertently changed.